### PR TITLE
fix e2e DNS issues and increase parallelism level by default

### PR DESCRIPTION
**Pull request by mmatczuk**

### CODE REVIEW

File: e2e/run.go [hunk 1]
```diff
@@ -31,7 +32,7 @@ var args = struct {
 	setup:    flag.String("setup", "", "Only run setups matching this regexp"),
 	run:      flag.String("run", "", "Only run tests matching this regexp"),
 	debug:    flag.Bool("debug", false, "Enables debug logs and preserves containers after running, this will run only the first matching setup"),
-	parallel: flag.Int("parallel", 1, "How many setups to run in parallel"),
+	parallel: flag.Int("parallel", runtime.NumCPU()/2+1, "How many setups to run in parallel"),
```
Comment (Choraden) [inline e2e/run.go:35]: Unfortunately, my Podman machine does not have access to all host's cpus. The parallelism becomes too high and some tests still fail.

Comment (Choraden) [reply e2e/run.go:35]: It's still DNS lookup error for services other than 'proxy'  🤔

**Comment by mmatczuk** [comment]
Updated DNS wait to all services.

**Comment by Choraden** [comment]
1. It looks like context cancellation leaks composes - they are not cleaned up. Let's fix it here or open an issue. 

2. Compose networks are visible for each other, thus they can't have the same name. Tests that use internal network fail when their setups happen to be executed simultaneously. Try `make -C e2e run-e2e SETUP="flag-dns"`

Other than that LGTM.

**Comment by mmatczuk** [comment]
I'm aware of the issues.

**Review by Choraden — APPROVED** [review]

**Pull request merged**